Library/Homebrew/extend/ENV/super.rb

```diff
@@ -67,6 +67,11 @@ def setup_build_environment(formula = nil)
     self["HOMEBREW_INCLUDE_PATHS"] = determine_include_paths
     self["HOMEBREW_LIBRARY_PATHS"] = determine_library_paths
 
+    if MacOS::Xcode.without_clt?
+      self["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version.to_s
+      self["SDKROOT"] = MacOS.sdk_path
```

Comment:
We should probably handle this elsewhere, earlier in the build process. When we reach here we're at a point where the build is already guaranteed to fail (and would have failed before this patch too).